testApp/src/App.svelte

```diff
@@ -0,0 +1,122 @@
+<!-- App.svelte -->
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import Gallery from './Components/Gallery.svelte';
+	import MovieDetail from './Components/MovieDetail.svelte';
+	import NavBar from './Components/NavBar.svelte';
+	import '/global.css';
+	import type { Movie } from '/Interfaces/Movie';
+	import type { Franchise } from '/Interfaces/Franchise';
+
+	export let bearer: string;
+
+	let movies: Movie[] = [];
+	let currentIndex: number = -1;
+	let showDetail: boolean = false;
+	let currentMovie: Movie = null;
+	let franchiseId: number = 17; // Default to Star Wars
+
+	async function fetchFranchiseData(id: number) {
```

Comment:
None of your functions seem to have a return type. As this is TypeScript, it would be very useful to use return types.